python/pyspark/sql/dataframe.py

```diff
@@ -3198,7 +3198,7 @@ def writeTo(self, table: str) -> DataFrameWriterV2:
         """
         return DataFrameWriterV2(self, table)
 
-    def to_pandas_on_spark(
+    def pandas_api(
```

Comment:
We can't just rename the API because we already have a release.
We should keep or deprecate the original function and add the new API.

Reply:
Makes sense! Deprecated `to_pandas_on_spark`.